### PR TITLE
chore: bump datafusion version to fix `last_value` regression

### LINKED ISSUE #4096: last_value()/first_value() returns incorrect and unstable result
### What type of bug is this?

Incorrect result

### What subsystems are affected?

Standalone mode

### Minimal reproduce step

```sql
CREATE TABLE monitor (
  host STRING,
  ts TIMESTAMP DEFAULT CURRENT_TIMESTAMP() TIME INDEX,
  cpu FLOAT64 DEFAULT 0,
  memory FLOAT64,
  PRIMARY KEY(host)) ENGINE=mito;

INSERT INTO monitor
VALUES
    ("127.0.0.1", 1, 0.5, 0.9),
    ("127.0.0.2", 2, 0.6, 0.8),
    ("127.0.0.3", 3, 0.7, 0.7),
    ("127.0.0.4", 4, 0.8, 0.6),
    ("127.0.0.4", 5, 0.9, 0.5),
    ("127.0.0.5", 6, 1.0, 0.4);

select last_value(cpu order by ts) from monitor;

select first_value(cpu order by ts) from monitor;
```

### What did you expect to see?

```
select last_value(cpu order by ts) from monitor;

+--------------------------------------------------------------+
| last_value(monitor.cpu) ORDER BY [monitor.ts ASC NULLS LAST] |
+--------------------------------------------------------------+
|                                                            1 |
+--------------------------------------------------------------+

select first_value(cpu order by ts) from monitor;

+---------------------------------------------------------------+
| first_value(monitor.cpu) ORDER BY [monitor.ts ASC NULLS LAST] |
+---------------------------------------------------------------+
|                                                           0.5 |
+---------------------------------------------------------------+
```

### What did you see instead?

```sql
select last_value(cpu order by ts) from monitor;
+--------------------------------------------------------------+
| last_value(monitor.cpu) ORDER BY [monitor.ts ASC NULLS LAST] |
+--------------------------------------------------------------+
|                                                          0.6 |
+--------------------------------------------------------------+

select first_value(cpu order by ts) from monitor;
+---------------------------------------------------------------+
| first_value(monitor.cpu) ORDER BY [monitor.ts ASC NULLS LAST] |
+---------------------------------------------------------------+
|                                                           0.6 |
+---------------------------------------------------------------+
```

### What operating system did you use?

Unrelated

### What version of GreptimeDB did you use?

0.8.1

### Relevant log output and stack trace

```bash
NA
```

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

... to the datafusion commit with PR https://github.com/apache/datafusion/pull/10783. 

Now `last_value` with string arg works:

```text
mysql> select last_value('3');
+-----------------------+
| last_value(Utf8("3")) |
+-----------------------+
| 3                     |
+-----------------------+
1 row in set (0.04 sec)
```

Also fix #4096 :

```text
mysql> CREATE TABLE monitor (
    ->   host STRING,
    ->   ts TIMESTAMP DEFAULT CURRENT_TIMESTAMP() TIME INDEX,
    ->   cpu FLOAT64 DEFAULT 0,
    ->   memory FLOAT64,
    ->   PRIMARY KEY(host)) ENGINE=mito;
Query OK, 0 rows affected (0.04 sec)

mysql> INSERT INTO monitor
    -> VALUES
    ->     ("127.0.0.1", 1, 0.5, 0.9),
    ->     ("127.0.0.2", 2, 0.6, 0.8),
    ->     ("127.0.0.3", 3, 0.7, 0.7),
    ->     ("127.0.0.4", 4, 0.8, 0.6),
    ->     ("127.0.0.4", 5, 0.9, 0.5),
    ->     ("127.0.0.5", 6, 1.0, 0.4);
Query OK, 6 rows affected (0.01 sec)

mysql> select last_value(cpu order by ts) from monitor;
+--------------------------------------------------------------+
| last_value(monitor.cpu) ORDER BY [monitor.ts ASC NULLS LAST] |
+--------------------------------------------------------------+
|                                                            1 |
+--------------------------------------------------------------+
1 row in set (0.05 sec)

mysql> select first_value(cpu order by ts) from monitor;
+---------------------------------------------------------------+
| first_value(monitor.cpu) ORDER BY [monitor.ts ASC NULLS LAST] |
+---------------------------------------------------------------+
|                                                           0.5 |
+---------------------------------------------------------------+
1 row in set (0.03 sec)
```

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/datatypes/src/value.rs [hunk 1]
```diff
@@ -877,12 +877,7 @@ impl TryFrom<ScalarValue> for Value {
             ScalarValue::Decimal128(v, p, s) => v
                 .map(|v| Value::Decimal128(Decimal128::new(v, p, s)))
                 .unwrap_or(Value::Null),
-            ScalarValue::Decimal256(_, _, _)
-            | ScalarValue::Struct(_)
-            | ScalarValue::FixedSizeList(_)
-            | ScalarValue::LargeList(_)
-            | ScalarValue::Dictionary(_, _)
-            | ScalarValue::Union(_, _, _) => {
+            _ => {
```
Comment (killme2008) [inline src/datatypes/src/value.rs:880]: According to our code style, it's better to list all the variants here.

File: src/datatypes/src/vectors/helper.rs [hunk 2]
```diff
@@ -234,12 +234,7 @@ impl Helper {
                 let vector = Decimal128Vector::from(vec![v]).with_precision_and_scale(p, s)?;
                 ConstantVector::new(Arc::new(vector), length)
             }
-            ScalarValue::Decimal256(_, _, _)
-            | ScalarValue::Struct(_)
-            | ScalarValue::FixedSizeList(_)
-            | ScalarValue::LargeList(_)
-            | ScalarValue::Dictionary(_, _)
-            | ScalarValue::Union(_, _, _) => {
+            _ => {
```
Comment (killme2008) [inline src/datatypes/src/vectors/helper.rs:237]: The same suggestion as above.

**Review by killme2008 — APPROVED** [review]
LGTM

**Comment by killme2008** [comment]
It's better to add a sqlness test to https://github.com/GreptimeTeam/greptimedb/blob/main/tests/cases/standalone/common/function/expression.sql

**Comment by fengjiachun** [comment]
This PR also fixed #4096

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**